src/main/java/racingcar/service/GamePlayService.java

```diff
@@ -0,0 +1,65 @@
+package racingcar.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import racingcar.dao.GameDao;
+import racingcar.dao.PlayerResultDao;
+import racingcar.domain.Car;
+import racingcar.domain.Cars;
+import racingcar.domain.RacingGame;
+import racingcar.domain.RandomNumberGenerator;
+import racingcar.dto.request.GameRequestDto;
+import racingcar.dto.request.GameSaveDto;
+import racingcar.dto.request.PlayerResultSaveDto;
+import racingcar.dto.response.GameResponseDto;
+import racingcar.dto.response.PlayerResultDto;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class GamePlayService {
+
+    private final GameDao gameDao;
+    private final PlayerResultDao playerResultDao;
+
+    public GamePlayService(GameDao gameDao, PlayerResultDao playerResultDao) {
+        this.gameDao = gameDao;
+        this.playerResultDao = playerResultDao;
+    }
+
+    @Transactional
+    public GameResponseDto playGame(GameRequestDto gameRequestDto) {
+        List<String> carNames = List.of(gameRequestDto.getNames().split(",", -1));
+        Cars cars = makeCars(carNames);
+        RacingGame racingGame = new RacingGame(cars, gameRequestDto.getCount());
+        racingGame.race(new RandomNumberGenerator());
+
+        List<String> winners = cars.calculateWinners();
+
+        Long gameId = createGame(String.join(",", winners), gameRequestDto.getCount());
+        List<PlayerResultSaveDto> playerResults = createPlayerResults(cars, gameId);
+        playerResultDao.savePlayerResult(playerResults);
+        List<PlayerResultDto> playerResultsByGameId = playerResultDao.findPlayerResultsByGameId(gameId);
+        return GameResponseDto.of(winners, playerResultsByGameId);
+    }
+
+    private Cars makeCars(List<String> carNames) {
+        List<Car> baseCars = carNames.stream()
+                .map(Car::new)
+                .collect(Collectors.toList());
+        return new Cars(baseCars);
+    }
+
+    private Long createGame(String winners, int trialCount) {
+        GameSaveDto gameSaveDto = new GameSaveDto(winners, trialCount);
+        Long gameId = gameDao.createGame(gameSaveDto);
+        return gameId;
+    }
```

Comment:
```suggestion
    private Long createGame(String winners, int trialCount) {
        GameSaveDto gameSaveDto = new GameSaveDto(winners, trialCount);
        return gameDao.createGame(gameSaveDto);
    }
```
불필요한 변수 할당을 할 필요는 없겠네요 😉

Reply:
나름 가독성을 위해 저렇게 작성했는데 없는게 더 가독성이 좋네요 ㅎ.. 수정했습니다!!

```
private Position getMaxPosition() {
    return cars.stream()
            .map(Car::getCurrentPosition)
            .max(Comparator.comparingInt(Position::getPosition))
            .orElseGet(Position::new);
}
```
위 코드처럼 stream을 도는 경우도 바로 return을 하는게 가독성이 더 좋다고 생각하시나요? stream에 대해 안다면 가독성이 괜찮은거 같은데, 모르는 사람이 보면 가독성이 좋지 않다고 생긱이 됩니다..!

Reply:
당연히 팀의 상황에 따라 다를 것 같아요! 개인적으로는 바로 반환하는 걸 선호하는 편입니다.
다만 레오가 고민하신 것처럼 "그렇지 않은게 가독성이 더 높아" 라고 한다면 과감하게 변수를 사용합니다. 👍